reframe/core/containers.py

```diff
@@ -119,32 +116,45 @@ class Sarus(ContainerPlatform):
     #: :default: :class:`False`
     with_mpi = fields.TypedField(bool)
 
+    #: Skip pull of images
```

Comment:
This violates the reproducibility; thus, not a keeper.
But (being devil's advocate), an image name like `qnib/spack-gromacs:latest` violates reproducibility as well. :)
To make an image reproducibile you would want to specify an image using the content hash.
```
docker image inspect qnib/spack-gromacs-shp2:c5n_18xl_off |jq '.[]|.RepoDigests'
[
  "qnib/spack-gromacs-shp2@sha256:4f5ae272e343395de2259e2745ab24f20a34688343f9f233cec895915f1704f5"
]
```
and not `qnib/spack-gromacs-shp2:c5n_18xl_off`

Reply:
That might be a thing btw. inspecting the actual hash and putting it into the job output as an strict description.
For some images like `ubuntu:18.04` you actually specify a manifest list (image index) and not a 'real' image.